pkg/controller/core/workload_controller.go

```diff
@@ -134,6 +135,17 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 {
```

Comment:
This is good by itself, but we should also make sure that a job cannot be unsuspended when the workload is finished.
If it's not already the case, you can do it in a follow up.